src/compiler/checker.ts

```diff
@@ -9592,7 +9591,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                         jsdocType = getAnnotatedTypeForAssignmentDeclaration(jsdocType, expression, symbol, declaration);
                     }
                     if (!jsdocType) {
-                        (types || (types = [])).push((isBinaryExpression(expression) || isCallExpression(expression)) ? getInitializerTypeFromAssignmentDeclaration(symbol, resolvedSymbol, expression, kind) : neverType);
+                        (types ??= []).push((isBinaryExpression(expression) || isCallExpression(expression)) ? getInitializerTypeFromAssignmentDeclaration(symbol, resolvedSymbol, expression, kind) : neverType);

```

Comment:
If we were changing this, I'd use `append` for efficiency, but, I think this change is no longer needed at this point either.